### ISSUE TITLE
make net.params private attribute

### ISSUE THREAD

**Issue by jasmainak** [jonescompneurolab/hnn-core#307]
Here is the code to reproduce:

```py
import os.path as op

import hnn_core
from hnn_core import simulate_dipole, read_params, Network
from hnn_core.viz import plot_dipole

hnn_core_root = op.dirname(hnn_core.__file__)

params_fname = op.join(hnn_core_root, 'param', 'default.json')
params = read_params(params_fname)

params['tstop'] = 30
net = Network(params, add_drives_from_params=True)
dpls = simulate_dipole(net, n_trials=1, postproc=True)

net.params['tstop'] = 20.

dpls = simulate_dipole(net, n_trials=1, postproc=True)
```

The issue is somewhere with `cell_response.times` which doesn't appear to be reset. Is this an abuse of the API? Or should this work? I am thinking that `net.params` should become private and `tstop` should be set from `simulate_dipole` only. Thoughts @rythorpe @ntolley ?

**Comment by ntolley** [comment]
I'd argue this is API misuse since we're actively discouraging direct modification of `params`. We've eliminated all modification of params in the examples (we do print the contents of it though). 

Also the reason it doesn't reset is because `cell_response.times` is only created during instantiation of the `Network`. I definitely support making `params` private since it is only used once to create the Network. The benefit of printing the params directly will soon become redundant as we develop functions to visualize connectivity information.

**Comment by jasmainak** [comment]
okay sounds good. Let's make `net.params` private then! I renamed the issue accordingly

**Comment by jasmainak** [comment]
@alexrockhill want to give this a shot if you have some free time? :)

**Comment by alexrockhill** [comment]
Love to contribute, either later tonight or tomorrow I'll give it a shot

**Comment by cjayb** [comment]
Apparently 'Addresses' didn't automatically close this related issue (#310)

@jasmainak are you satisfied with being called an API-bully? ;) I'm actually a little surprised about this...

Since #291 the `cell_response` [should have been reset](https://github.com/jonescompneurolab/hnn-core/blob/22c5df231074aac5a2420668e3872fa3d5b74c32/hnn_core/dipole.py#L96) by `simulate_dipole`, right?

**Comment by jasmainak** [comment]
hahaha :) yep, but [`cell_response.reset`](https://github.com/jonescompneurolab/hnn-core/blob/22c5df231074aac5a2420668e3872fa3d5b74c32/hnn_core/cell_response.py#L218) does not reset `cell_response.times` which is in turn used to [instantiate the dipole vector](https://github.com/jonescompneurolab/hnn-core/blob/22c5df231074aac5a2420668e3872fa3d5b74c32/hnn_core/network_builder.py#L293). That fails with a [size mismatch](https://github.com/jonescompneurolab/hnn-core/blob/22c5df231074aac5a2420668e3872fa3d5b74c32/hnn_core/network_builder.py#L477) in the next simulation. I'm wondering if `cell_response.times` should also be reset to avoid such issues in the future.

By the way, I found this issue when working on #77 ... the more you use the software, the more issues you'll find and help tighten the API :)

**Comment by jasmainak** [comment]
closed by #310

**Comment by cjayb** [comment]
Ah yes, times, duh. Totally agree about your comment on actual use being the best test of the API!
